themes/classic/_dev/js/listing.js

```diff
@@ -127,20 +127,24 @@ $(document).ready(() => {
 
   $('body').on('change', '#search_filters input[data-search-url]', function (event) {
     prestashop.emit('updateFacets', parseSearchUrl(event));
+    window.scrollTo(0, 0);
   });
 
   $('body').on('click', '.js-search-filters-clear-all', function (event) {
     prestashop.emit('updateFacets', parseSearchUrl(event));
+    window.scrollTo(0, 0);
   });
 
   $('body').on('click', '.js-search-link', function (event) {
     event.preventDefault();
     prestashop.emit('updateFacets', $(event.target).closest('a').get(0).href);
+    window.scrollTo(0, 0);
   });
 
   $('body').on('change', '#search_filters select', function (event) {
     const form = $(event.target).closest('form');
     prestashop.emit('updateFacets', '?' + form.serialize());
+    window.scrollTo(0, 0);
   });
 
   prestashop.on('updateProductList', (data) => {
```

Comment:
What do you think about having `window.scrollTo(0, 0);` inside `prestashop.on('updateProductList', (data) => {` instead?

Reply:
I can absolutely do that instead. However, the bahaviour will be _slightly_ different.

* Right now, it will render the change (i.e. filter, sort order, pagination, etc) and _then_ scroll to the top.
* If inside the event, it will scroll to the top as soon as possible.

What do you think?

Reply:
I understand your concern, but does it change from the original version?

Let's ping for @TristanLDD about this issue, but if this is the same behavior from the core assets, I would be fine with it.

Reply:
It's essentially the same thing, the only thing that differs is _when_ the scroll occurs. After or before the change is set.

One plus on having it the event is DRY code though.